github/Enterprise.py

```diff
@@ -0,0 +1,66 @@
+############################ Copyrights and license ############################
+#                                                                              #
+# Copyright 2023 Yugo Hino <henom06@gmail.com>                                 #
+#                                                                              #
+# This file is part of PyGithub.                                               #
+# http://pygithub.readthedocs.io/                                              #
+#                                                                              #
+# PyGithub is free software: you can redistribute it and/or modify it under    #
+# the terms of the GNU Lesser General Public License as published by the Free  #
+# Software Foundation, either version 3 of the License, or (at your option)    #
+# any later version.                                                           #
+#                                                                              #
+# PyGithub is distributed in the hope that it will be useful, but WITHOUT ANY  #
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS    #
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more #
+# details.                                                                     #
+#                                                                              #
+# You should have received a copy of the GNU Lesser General Public License     #
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+from typing import Any, Dict
+
+from github.EnterpriseConsumedLicenses import EnterpriseConsumedLicenses
+from github.GithubObject import Attribute, CompletableGithubObject, NotSet
+
+
+class Enterprise(CompletableGithubObject):
+    """
+    This class represents Enterprises. The reference can be found here https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/license#list-enterprise-consumed-licenses
+    """
+
```

Comment:
The thing with object `Enterprise` is that it does not exist w.r.t. the Github API. We merely have this class to collect all endpoints hat start with `/enterprise/{enterprise}` under one class.

Firstly, this is `NonCompletableGithubObject`, secondly, we create it from a requester and the enterprise name:
```suggestion
class Enterprise(NonCompletableGithubObject):
    """
    This class represents Enterprises. Such objects do not exist in the Github API, so this class merely collects all endpoints the start with /enterprise/{enterprise}/. See methods below for specific endpoints and docs.
    """

    def __init__(
        self,
        requester: "Requester",
        slug_or_id: str,
    ):
        super().__init__(requester, {}, {"slug_or_id": slug_or_id, "url": s"/enterprise/{slug_or_id}"}, True)

```

Reply:
Thank you. Fixed it.